client-js/app/scripts/places.ts

```diff
@@ -1,19 +1,35 @@
 // Function for creating URLs of all the places in the application.
 
-export function grapherUrl(path: Array<string>, dir?: string): string {
-  return '/grapher/' + pathStr(path, dir);
-}
+export class Places {
+  fullPrefix: string;
```

Comment:
The idea is that we create an instance of the `Places` class and then pass that around to any parts of the app that need to generate URLs to link to other places within the app. In order to generate these URLs correctly, you need to know which manager you're talking to, and possibly what url prefix the app is being served under. For example, we might be on a shared host where scalabrad-web is not at the root like `https://shared-host.com/` but rather at `https://shared-host.com/labrad/`. Then all intra-app links need to have that `/labrad` prefix, so we set that when creating the `Places` instance and then the other parts of the app don't need to change to create the correct URLs.